install/pre-install-payload/payload.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+script_dir=$(dirname "$(readlink -f "$0")")
+
+containerd_repo=${containerd_repo:-"https://github.com/confidential-containers/containerd"}
+containerd_version=${containerd_version:-"v1.6.6.0"}
+containerd_dir="$(mktemp -d -t containerd-XXXXXXXXXX)/containerd"
+
+SNAPSHOTTER=${SNAPSHOTTER:-"nydus"}
+nydus_snapshotter_repo=${nydus_snapshotter_repo:-"https://github.com/containerd/nydus-snapshotter"}
+nydus_snapshotter_version=${nydus_snapshotter_version:-"main"}
+nydus_snapshotter_dir="$(mktemp -d -t nydus_snapshotter-XXXXXXXXXX)/nydus_snapshotter"
+
+registry="${registry:-quay.io/confidential-containers}"
+
+supported_arches=(
+	"linux/amd64"
+	"linux/s390x"
+)
+
+function setup_env_for_arch() {
+	case "$1" in
+	"linux/amd64")
+		kernel_arch="x86_64"
+		golang_arch="amd64"
+		;;
+	"linux/s390x")
+		kernel_arch="s390x"
+		golang_arch="s390x"
+		;;
+	*) echo "$1 is not supported" >/dev/stderr && exit 1 ;;
+	esac
+}
+
+function build_payload_image() {
+	local version=$1
+	local repo=$2
+	local dir=$3
+	local arch=$4
+	local target_registry=$5
+
+	setup_env_for_arch "$arch"
+
+	echo "Building payload image for $arch"
+	docker buildx build \
+		--build-arg VERSION="$version" \
+		--build-arg URL="$repo" \
+		--build-arg ARCH="$golang_arch" \
+		-f "$dir/Dockerfile" \
+		-t "${target_registry}:${kernel_arch}-${tag}" \
+		--platform="$arch" \
+		--load \
+		.
+	docker push "${target_registry}:${kernel_arch}-${tag}"
+}
+
+function build_containerd_payload() {
+	for arch in "${supported_arches[@]}"; do
+		build_payload_image "$containerd_version" "$containerd_repo" "containerd" "$arch" "$registry/container-engine-for-cc-payload"
+	done
+	create_manifest "$registry/container-engine-for-cc-payload"
+}
+
+function build_snapshotter_payload() {
+	case "$SNAPSHOTTER" in
+	nydus)
+		for arch in "${supported_arches[@]}"; do
+			build_payload_image "$nydus_snapshotter_version" "$nydus_snapshotter_repo" "remote-snapshotter/nydus-snapshotter" "$arch" "$registry/remote-snapshotter-for-cc-payload-$SNAPSHOTTER"
+		done
+		create_manifest "$registry/remote-snapshotter-for-cc-payload-$SNAPSHOTTER"
+		;;
+	tardev)
+		# TODO
+		;;
+	*) echo "$SNAPSHOTTER is not supported" >/dev/stderr && exit 1 ;;
+	esac
+}
+
+function create_manifest() {
+	local target_registry=$1
+	docker manifest create \
+		"${target_registry}:${tag}" \
+		--amend "${target_registry}:x86_64-${tag}" \
+		--amend "${target_registry}:s390x-${tag}"
+
+	docker manifest create \
+		"${target_registry}:latest" \
+		--amend "${target_registry}:x86_64-${tag}" \
+		--amend "${target_registry}:s390x-${tag}"
+
+	docker manifest push "${target_registry}:${tag}"
+	docker manifest push "${target_registry}:latest"
+}
+
+function main() {
+	tag=$(git rev-parse HEAD)
+	case $1 in
+	containerd) build_containerd_payload ;;
+	snapshotter)
+		build_snapshotter_payload
+		;;
```

Comment:
I've re-worked the pre-install build, so it can be made more generic, as we **do** need to provide **the very same image** for the patched containerd replacement, for the snapshotter setup, and for whatever else may come in the future.

I am locally cherry-picking, adapting, and rebasing your patches atop of https://github.com/confidential-containers/operator/pull/253, as some ground work was needed.